### PR TITLE
fix(agw): Fixed and reenabled enb complete reset scenario

### LINKED ISSUE #12583: The test case enb complete reset is failing
### Your Environment

- **Version:** Local Development Setup
- **Affected Component:** Access Gateway
- **Affected Subcomponent:** Magma MME/SPGW service
- **Deployment Environment:** Vagrant (AGW)

### Setup Details
The S1APTester ([https://github.com/facebookexperimental/S1APTester](https://github.com/facebookexperimental/S1APTester)) was used along with Magma EPC in Vagrant based setup to reproduce the issue.

### Describe the Issue
The test case s1aptests/test_enb_complete_reset.py is failing on latest master with assert because the test case is expecting detach accept indication which is not received.

The failure terminal log is as follows:
```
F
======================================================================
FAIL: ENB Complete Reset:
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vagrant/magma/lte/gateway/python/integ_tests/s1aptests/test_enb_complete_reset.py", line 90, in test_enb_complete_reset
    wait_for_s1_ctxt_release=True,
  File "/home/vagrant/magma/lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py", line 435, in detach
    == response.msg_type
AssertionError:
```

As per the packet capture, without detach accept being sent, UE context release command is sent with cause unknown mme-ue-s1ap-id. Another issue observed is that the Detach request is sent in UL NAS Transport instead of initial UE message.

image

From the AGW syslog, MME can be seen crashing and restarting with following traceback:
```
Apr 28 11:05:39 magma-dev-focal mme[1155170]: /home/vagrant/magma/lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_handlers.c:1407:27: runtime error: member access within null pointer of type 'struct ue_description_t'
Apr 28 11:05:39 magma-dev-focal mme[1155170]: AddressSanitizer:DEADLYSIGNAL
Apr 28 11:05:39 magma-dev-focal mme[1155170]: =================================================================
Apr 28 11:05:39 magma-dev-focal mme[1155170]: ==1155170==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x555dad3179ef bp 0x7f87e11c3360 sp 0x7f87e11c3130 T21)
Apr 28 11:05:39 magma-dev-focal mme[1155170]: ==1155170==The signal is caused by a WRITE memory access.
Apr 28 11:05:39 magma-dev-focal mme[1155170]: ==1155170==Hint: address points to the zero page.
Apr 28 11:05:39 magma-dev-focal sctpd[1155050]: I0428 11:05:39.314790 1155447 sctp_connection.cpp:205] Unhandled notification type 32777
Apr 28 11:05:39 magma-dev-focal mme[1155170]:     #0 0x555dad3179ee in s1ap_mme_generate_ue_context_release_command /home/vagrant/magma/lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_handlers.c:1407
Apr 28 11:05:39 magma-dev-focal mme[1155170]:     #1 0x555dad37ba3c in s1ap_mme_handle_uplink_nas_transport /home/vagrant/magma/lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_nas_procedures.c:336
Apr 28 11:05:39 magma-dev-focal mme[1155170]:     #2 0x555dad300f1f in s1ap_mme_handle_message /home/vagrant/magma/lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_handlers.c:235
Apr 28 11:05:39 magma-dev-focal mme[1155170]:     #3 0x555dad0d23c0 in handle_message /home/vagrant/magma/lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme.c:138
Apr 28 11:05:39 magma-dev-focal mme[1155170]:     #4 0x7f87f1b545c6 in zloop_start (/lib/x86_64-linux-gnu/libczmq.so.4+0x295c6)
Apr 28 11:05:39 magma-dev-focal mme[1155170]:     #5 0x555dad0d3fbb in s1ap_mme_thread /home/vagrant/magma/lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme.c:317
Apr 28 11:05:39 magma-dev-focal mme[1155170]:     #6 0x7f87f261e608 in start_thread /build/glibc-eX1tMB/glibc-2.31/nptl/pthread_create.c:477
Apr 28 11:05:39 magma-dev-focal mme[1155170]:     #7 0x7f87f0aa9292 in __clone (/lib/x86_64-linux-gnu/libc.so.6+0x122292)
Apr 28 11:05:39 magma-dev-focal mme[1155170]: AddressSanitizer can not provide additional info.
Apr 28 11:05:39 magma-dev-focal mme[1155170]: SUMMARY: AddressSanitizer: SEGV /home/vagrant/magma/lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_handlers.c:1407 in s1ap_mme_generate_ue_context_release_command
Apr 28 11:05:39 magma-dev-focal mme[1155170]: Thread T21 created by T0 here:
Apr 28 11:05:39 magma-dev-focal mme[1155170]:     #0 0x7f87f287e805 in pthread_create (/lib/x86_64-linux-gnu/libasan.so.5+0x3a805)
Apr 28 11:05:39 magma-dev-focal mme[1155170]:     #1 0x555db10a3046 in itti_create_task /home/vagrant/magma/lte/gateway/c/core/oai/lib/itti/intertask_interface.c:333
Apr 28 11:05:39 magma-dev-focal mme[1155170]:     #2 0x555dad0d46ed in s1ap_mme_init /home/vagrant/magma/lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme.c:342
Apr 28 11:05:39 magma-dev-focal mme[1155170]:     #3 0x555dad059c93 in main /home/vagrant/magma/lte/gateway/c/core/oai/oai_mme/oai_mme.c:151
Apr 28 11:05:39 magma-dev-focal mme[1155170]:     #4 0x7f87f09ae0b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)
Apr 28 11:05:39 magma-dev-focal mme[1155170]: ==1155170==ABORTING
```

The MME crash and sending of detach request in initial UE message should be fixed independently for this issue.

**To Reproduce**
1) Run the test case individually

### PR DESCRIPTION
**Pull request by VinashakAnkitAman**
## Title
fix(agw): Fixed and reenabled enb complete reset scenario

## Summary
As per the Github issue #12583, the enb complete reset testcase was failing. The issue was on the S1APTester application and has been fixed as part of https://github.com/magma/S1APTester/pull/81. This PR reenables the test case as part of the sanity

## Test plan
Verified with sanity

Signed-off-by: VinashakAnkitAman <ankit.aman@radisys.com>

### CODE REVIEW

File: lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py [hunk 1]
```diff
@@ -84,8 +84,8 @@ class S1ApUtil(object):
 
     _cond = threading.Condition()
     _msg = Queue()
-    # Default maximum wait time is 180 sec (3 min)
-    MAX_RESP_WAIT_TIME = 180
+    # Default maximum wait time is 60 sec (1 min)
+    MAX_RESP_WAIT_TIME = 60
```
Comment (pruthvihebbani) [inline lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py]: Are these changes needed as part of the fix?

File: lte/gateway/python/integ_tests/s1aptests/util/traffic_util.py [hunk 2]
```diff
@@ -32,7 +32,7 @@
 )
 
 # Tests shouldn't take longer than a few minutes
-TRAFFIC_TEST_TIMEOUT_SEC = 180
+TRAFFIC_TEST_TIMEOUT_SEC = 60
```
Comment (pruthvihebbani) [inline lte/gateway/python/integ_tests/s1aptests/util/traffic_util.py]: same as above

**Review by pruthvihebbani — CHANGES_REQUESTED** [review]

File: lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py (hunk 1 above)
Comment (VinashakAnkitAman) [reply lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py]: Removed the changes and will raise a separate PR for this

File: lte/gateway/python/integ_tests/s1aptests/util/traffic_util.py (hunk 2 above)
Comment (VinashakAnkitAman) [reply lte/gateway/python/integ_tests/s1aptests/util/traffic_util.py]: Removed the changes and will raise a separate PR for this

**Review by rsarwad — APPROVED** [review]
LGTM

**Review by pruthvihebbani — APPROVED** [review]

**Comment by pruthvihebbani** [comment]
I am unable to merge as mme test is failing. Can you rebase?
